api/src/main/java/org/apache/iceberg/types/Types.java

```diff
@@ -81,6 +82,16 @@ public static PrimitiveType fromPrimitiveString(String typeString) {
     throw new IllegalArgumentException("Cannot parse type string to primitive: " + typeString);
   }
 
+  @Deprecated
```

Comment:
Since this has a different return type, it makes sense to me that it would be supported. I think the motivation to deprecate it is that we want uses to move to `fromString` / `fromTypeName`, but we can do that by searching for uses and updating.